src/Documents/DocumentStore.ts

```diff
@@ -85,14 +85,42 @@ export class DocumentStore implements IDocumentStore {
       .then((): IDocumentStore => this);
   }
 
-  public openSession(database?: string, forceReadFromMaster: boolean = false): IDocumentSession {
+  public async openSession(transaction: ITransaction) : Promise<void>;
+  public async openSession(options: ISessionOptions, transaction: ITransaction) : Promise<void>;
+  public openSession(database?: string, forceReadFromMaster?: boolean) : IDocumentSession;
+  public openSession(firstArg?: string | ITransaction | ISessionOptions, secondArg?: ITransaction | boolean): IDocumentSession | Promise<void> {
```

Comment:
Optional features are a mess. And I would rather not start a new client with already obselete features.

Returning the hilo is optional, so we can skip it if needed but I agree that this is a good idea.
However, this is on the _store_ level, which isn't typically used with a `with` / `using`.

Typically you create it once and use if for the lifetime of your app